cmake/modules/MaterialXConfig.cmake.in

```diff
@@ -0,0 +1,29 @@
+#
+# TM & (c) 2021 Lucasfilm Entertainment Company Ltd. and Lucasfilm Ltd.
+# All rights reserved.  See LICENSE.txt for license.
+#
+# MaterialX CMake configuration file. Mostly auto-generated.
+#
+
+# Auto-generated content:
+@PACKAGE_INIT@
+
+# Gather MaterialX targets:
+include("${CMAKE_CURRENT_LIST_DIR}/@CMAKE_PROJECT_NAME@Targets.cmake")
+
+# Resource paths:
+# MATERIALX_BASE_DIR         MaterialX root installation directory
+# MATERIALX_STDLIB_DIR       Path to the MaterialX standard library directory
+# MATERIALX_PBRLIB_DIR       Path to the MaterialX pbr library directory
+# MATERIALX_BXDFLIB_DIR      Path to the Surface shaders library directory e.g. standard surface
+# MATERIALX_PYTHON_DIR       Path to MaterialX Python library
+# MATERIALX_RESOURCES_DIR    Path to MaterialX Resources (sample data, mtlx etc)
+
+set_and_check(MATERIALX_BASE_DIR "@PACKAGE_CMAKE_INSTALL_PREFIX@")
+set_and_check(MATERIALX_STDLIB_DIR "@PACKAGE_CMAKE_INSTALL_PREFIX@/libraries/stdlib")
+set_and_check(MATERIALX_PBRLIB_DIR "@PACKAGE_CMAKE_INSTALL_PREFIX@/libraries/pbrlib")
+set_and_check(MATERIALX_BXDFLIB_DIR "@PACKAGE_CMAKE_INSTALL_PREFIX@/libraries/bxdf")
+set_and_check(MATERIALX_PYTHON_DIR "@PACKAGE_CMAKE_INSTALL_PREFIX@/python")
+set_and_check(MATERIALX_RESOURCES_DIR "@PACKAGE_CMAKE_INSTALL_PREFIX@/resources")
+
+check_required_components(@CMAKE_PROJECT_NAME@)
```

Comment:
Replaces obsolete FindMaterialX.cmake

Reply:
Read the docs, but don't claim to understand how this works but it seems for each exposed library directories we do extra checking.

I believe  FindMaterialX was written before we introduced ADSK library folders. I assume for this or for any new ones that the additional set_and_check() should be added ?

Reply:
The set_and_check makes sure the directory exists before setting the CMake variable. It ensures that the library is fully installed by checking that the expected paths are present on the filesystem.

We could probably add the adsk folder, but I think these also predate the fully recursive library loader. Maybe we need a single MATERIALX_LIBRARY_DIR and can remove the STDLIB and PBRLIB entries?

Reply:
This could work for "standard libraries" and just point to the root.